PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/FunctionDeploymentPanel.java

```diff
@@ -84,15 +248,118 @@ protected JLabel getLblMavenProject() {
+        configuration.setModule((Module) cbFunctionModule.getSelectedItem());
+        configuration.setDeploymentStagingDirectory(txtStagingFolder.getText());
+    }
+
+    private AppServicePlan getAppServicePlan(FunctionApp functionApp) {
+        if (!appServicePlanMap.containsKey(functionApp.id())) {
+            appServicePlanMap.put(functionApp.id(), functionApp.manager().appServicePlans().getById(functionApp.appServicePlanId()));
+        }
+        return appServicePlanMap.get(functionApp.id());
+    }
+
+    private void selectFunction() {
+        final Object value = cbxFunctionApp.getSelectedItem();
+        if (value != null && value instanceof ResourceEx) {
+            selectedFunctionApp = (ResourceEx<FunctionApp>) cbxFunctionApp.getSelectedItem();
+            presenter.loadAppSettings(selectedFunctionApp.getResource());
+        } else if (Comparing.equal(CREATE_NEW_FUNCTION_APP, value)) {
+            // Create new web app
+            cbxFunctionApp.setSelectedItem(null);
+            ApplicationManager.getApplication().invokeLater(() -> createNewFunctionApp());
+        }
+    }
 
+    private void createNewFunctionApp() {
+        // todo: add create function dialog
     }
 
     private void createUIComponents() {
-        // TODO: place custom component creation code here
+        lblCreateFunctionApp = new HyperlinkLabel(CREATE_NEW_FUNCTION);
+        lblCreateFunctionApp.addHyperlinkListener(e -> createNewFunctionApp());
+
+        final String localSettingPath = Paths.get(project.getBasePath(), "local.settings.json").toString();
+        appSettingsTable = new AppSettingsTable(localSettingPath);
+        pnlAppSettings = AppSettingsTableUtils.createAppSettingPanel(appSettingsTable);
+    }
+
+    class FunctionAppCombineBoxRender extends ListCellRendererWrapper {
+
+        private final JComboBox comboBox;
+        private final int cellHeight;
+        private static final String TEMPLATE_STRING = "<html><div>TEMPLATE</div><small>TEMPLATE</small></html>";
+
+        public FunctionAppCombineBoxRender(JComboBox comboBox) {
+            this.comboBox = comboBox;
+            final JLabel template = new JLabel(TEMPLATE_STRING);
+            //Create a multi-line jlabel and calculate its preferred size
+            this.cellHeight = template.getPreferredSize().height;
+        }
+
+        @Override
+        public void customize(JList list, Object value, int index, boolean b, boolean b1) {
+            if (value == null) {
+                return;
+            } else if (value instanceof String) {
+                setText(getStringLabelText((String) value));
+            } else {
+                final ResourceEx<FunctionApp> function = (ResourceEx<FunctionApp>) value;
+                // For label in combobox textfield, just show webapp name
+                final String text = index >= 0 ? getFunctionAppLabelText(function.getResource()) : function.getResource().name();
```

Comment:
what's the case for negative index?

Reply:
This is for Label in editor, or the editor will be also shown as multi-line